src/coreclr/src/debug/daccess/request.cpp

```diff
@@ -4500,3 +4500,165 @@ HRESULT ClrDataAccess::GetMethodsWithProfilerModifiedIL(CLRDATA_ADDRESS mod, CLR
 
     return hr;
 }
+
+HRESULT ClrDataAccess::GetGenerationTable(unsigned int cGenerations, struct DacpGenerationData *pGenerationData, unsigned int *pNeeded)
+{
+    if (cGenerations > 0 && pGenerationData == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    SOSDacEnter();
+
+    HRESULT hr = S_OK;
+    unsigned int numGenerationTableEntries = NUMBERGENERATIONS + 1;
+    if (pNeeded != NULL)
+    {
+        *pNeeded = numGenerationTableEntries;
+    }
+
+    if (cGenerations < numGenerationTableEntries)
+    {
+        hr = HRESULT_FROM_WIN32(ERROR_INSUFFICIENT_BUFFER);
+    }
+    else
+    {
+        for (unsigned int i=0; i < numGenerationTableEntries; i++)
+        {
+            DPTR(dac_generation) generation = GenerationTableIndex(g_gcDacGlobals->generation_table, i);
+            pGenerationData[i].start_segment = (CLRDATA_ADDRESS) dac_cast<TADDR>(generation->start_segment);
+
+            pGenerationData[i].allocation_start = (CLRDATA_ADDRESS) generation->allocation_start;
+
+            DPTR(gc_alloc_context) alloc_context = dac_cast<TADDR>(generation) + offsetof(dac_generation, allocation_context);
+            pGenerationData[i].allocContextPtr = (CLRDATA_ADDRESS)alloc_context->alloc_ptr;
+            pGenerationData[i].allocContextLimit = (CLRDATA_ADDRESS)alloc_context->alloc_limit;
+        }
+    }
+
+    SOSDacLeave();
+    return hr;
+}
+
+HRESULT ClrDataAccess::GetFinalizationFillPointers(unsigned int cFillPointers, CLRDATA_ADDRESS *pFinalizationFillPointers, unsigned int *pNeeded)
```

Comment:
Maybe I'm completely missing something, but doesn't it already support calling twice? It will only currently return E_INVALIDARG if `cFillPointers > 0 && pFinalizationFillPointers == NULL` so it should be valid to call it with `pFinalizationFillPointers == NULL` as long as `cFillPointers == 0`

Reply:
See message sent via teams, having trouble with GitHub 500 errors right now.